pandas/core/groupby/groupby.py

```diff
@@ -1119,7 +1120,24 @@ def _resolve_numeric_only(self, numeric_only: bool | lib.NoDefault) -> bool:
             # i.e. not explicitly passed by user
             if self.obj.ndim == 2:
                 # i.e. DataFrameGroupBy
-                numeric_only = True
+                # Checking if the dataframe has non-numeric features
```

Comment:
this is very complicated, see if you can do this in a simpler way

Reply:
ok , working on it

Reply:
I have updated a simpler version

Reply:
ill need to look at the linked issue more closely, but im very skeptical that this method is the right place for a fix.

Reply:
Explicitly numeric_only =False solves the error , so I looked into `_resolve_numeric_only` function.
It only checked for dataframe groupby object and set numeric_only to True , which forced the non_numeric columns out of the aggregation . That's why I included checks for multi datatype  dataframe .

Reply:
The _point_ of `numeric_only` is to force non-numeric columns out of the aggregation.  Changing the behavior here "fixed" the case you tested, but also changed a bunch of other behavior that broke a bunch of other tests

Reply:
OK I get ur part, I will check for numeric columns, if there are none only then it should set numeric_only to false, that should solve the issue and not violate the documentation

Reply:
I'm not sure you do.

`numeric_only` is an argument passed by the user, or defaulting to True if not explicitly passed.  We should _never_ be changing it based on what dtypes are actually present.

Reply:
But why can't the default value be set checking the datatypes , why does it have to be True always .